### PR TITLE
Update architecture diagrams in DeveloperGuide.md

### PR DESCRIPTION
**Pull request by KSHan29**
- Updated architecture diagrams to reflect our implementation as well as any method name changes
- Updated architecture diagrams documentation as well

### CODE REVIEW

File: docs/diagrams/LogicClassDiagram.puml [hunk 1]
```diff
@@ -6,7 +6,7 @@ skinparam classBackgroundColor LOGIC_COLOR
 
 package Logic {
 
-Class AddressBookParser
+Class NuSchedulerParser
 Class XYZCommand
```
Comment (ichigh0st) [inline docs/diagrams/LogicClassDiagram.puml:10]: there should be 2 classes: ZX commands and Y commands that inherit from Command

File: docs/diagrams/ParserClasses.puml [hunk 2]
```diff
@@ -19,20 +19,21 @@ Class Prefix
...
+HiddenOutside ..> NuSchedulerParser
 
-AddressBookParser .down.> XYZCommandParser: creates >
-
-XYZCommandParser ..> XYZCommand : creates >
-AddressBookParser ..> Command : returns >
-XYZCommandParser .up.|> Parser
-XYZCommandParser ..> ArgumentMultimap
-XYZCommandParser ..> ArgumentTokenizer
+NuSchedulerParser .down.> XCommandParser: creates >
```
Comment (ichigh0st) [inline docs/diagrams/ParserClasses.puml:24]: a ZXCommandParser should inherit from XCommandParser

File: docs/diagrams/StorageClassDiagram.puml [hunk 3]
```diff
@@ -29,15 +29,15 @@ HiddenOutside ..> Storage
...
+Storage -right-|> NuSchedulerStorage
 
 JsonUserPrefsStorage .up.|> UserPrefsStorage
-JsonAddressBookStorage .up.|> AddressBookStorage
-JsonAddressBookStorage ..> JsonSerializableAddressBook
-JsonSerializableAddressBook --> "*" JsonAdaptedPerson
-JsonAdaptedPerson --> "*" JsonAdaptedTag
+JsonNuSchedulerStorage .up.|> NuSchedulerStorage
+JsonNuSchedulerStorage ..> JsonSerializableNuScheduler
+JsonSerializableNuScheduler --> "*" JsonAdaptedProfile
```
Comment (ichigh0st) [inline docs/diagrams/StorageClassDiagram.puml:40]: could also include JsonAdaptedEvent

**Review by ichigh0st — COMMENTED** [review]
a few things to fixm otherwise nice updates!

File: docs/diagrams/LogicClassDiagram.puml (hunk 1 above)
Comment (KSHan29) [reply docs/diagrams/LogicClassDiagram.puml:10]: Thanks for the catches!

File: docs/diagrams/ParserClasses.puml (hunk 2 above)
Comment (KSHan29) [reply docs/diagrams/ParserClasses.puml:24]: I checked the code again and I think the `ZXCommandParser` also implements the `Parser` interface using generic type of `ZXCommand`. But for the `ZXCommand`, it does inherit `XCommand`. I will update the mistakes in the diagrams and DG, thanks for catching!

**Review by ichigh0st — APPROVED** [review]

**Pull request merged**